xla/service/gpu/gpu_memory_space_assignment.h

```diff
@@ -49,15 +45,37 @@ inline BufferAssigner::Colorer CollectiveColorer() {
         HloOpcode::kCollectivePermuteDone,
         HloOpcode::kAllToAll,
     };
+  if (kSupportedOpcodes->contains(alias->instruction()->opcode()))
+    return true;
+  return (alias->instruction()->opcode() == HloOpcode::kAsyncStart ||
+          alias->instruction()->opcode() == HloOpcode::kAsyncDone) &&
+             kSupportedOpcodes->contains(
+                 alias->instruction()->async_wrapped_opcode());
+}
+
+inline bool IsNvshmemOp(const HloValue* alias) {
```

Comment:
Is this limited to mosaic? If so, maybe include mosaic in the name. If not, can you add a comment what is considered an nvshmem op?

Reply:
This code currently looks for mosaic gpu custom call ops which have nvshmem instructions - I think the current comment explains this well? IF not, I can add more details.
With @Tixxx's host-side nvshmem collectives which are coming soon, we will also have to extend this to cover collective ops which use the nvshmem backend.